### PR TITLE
Persistent server in Go

### LINKED ISSUE #23: Persist server

### PR DESCRIPTION
**Pull request by alevy**
This implementation uses server sent events (https://html.spec.whatwg.org/multipage/server-sent-events.html) instead of WebSockets to deliver events to the client, but otherwise uses the same API. This also modifies the client's serverComm.js to use server-sent events.

Internally, the server stores mailboxes and one-time-keys in Pebble--a pure-Go embedded KV-store similar to RocksDB. Unfortunately, because serializable transactions aren't supported natively, the server takes coarse grain locks to do some of the updates, which could certainly be improved in future revisions.

Note: there is an outstanding bug in the Node version of the server that this PR doesn't fix (but it does fix it in the Go version) where `delete` messages expects a query parameter `seqID`, but the client sends it in a JSON body.

Closes #23

### CODE REVIEW

**Comment by nataliepopescu** [comment]
Starting to review now; how do I run the go server?

File: server/frida_server.go [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+	defer func() {
+		server.closingClients <- deviceId
+	}()
+
+	notify := req.Context().Done()
+
+	go func() {
+		<-notify
+		server.closingClients <- deviceId
+	}()
```
Comment (nataliepopescu) [inline server/frida_server.go:417]: What is this req.Context().Done() stuff for?

Comment (alevy) [reply server/frida_server.go:417]: `Context().Done()` is a channel that yields when the "context" closes, in other words, in this case, when the client disconnects.

File: core/client/serverComm.js [hunk 2]
```diff
@@ -94,9 +92,4 @@ export class ServerComm {
             return (await response.json())['otkey'];
         }
     }
-    disconnect() {
-        if (this.#socket) {
-            this.#socket.disconnect();
```
Comment (alevy) [inline core/client/serverComm.js:100]: Actually, we probably should call `this.#socket.close()` here

**Review by nataliepopescu — APPROVED** [review]

**Pull request merged**